lib/cn/cn_tree.c

```diff
@@ -821,7 +821,8 @@ cn_tree_view_create(struct cn *cn, struct table **view_out)
 
             kvset_get_ref(kvset);
             s->kvset = kvset;
-            s->node_loc = node->tn_loc;
+            s->nodeid = kvset_get_nodeid(kvset);
```

Comment:
I don't understand the reassignment here. Seems like you are just asserting that the kvset's nodeid is the same as the node's. Could you just do `assert(kvset_get_nodeid(kvset) == s->nodeid)`?